pkgs/build-support/docker/default.nix

```diff
@@ -684,6 +685,23 @@ rec {
     in
     result;
 
+  # Provide a /etc/passwd and /etc/group that contain root and nobody.
+  # Useful when packaging binaries that insist on using nss to look up
+  # username/groups (like nginx).
+  fakeNss = symlinkJoin {
+    name = "fake-nss";
+    paths = [
+      (writeTextDir "etc/passwd" ''
+        root:x:0:0:root user:/root:/bin/sh
```

Comment:
That shell entry is likely bogus as not every container has `/bin/sh`. Same is true for the home directories of the two users.

Reply:
What about `/var/empty` as homedir for both?

I'd like to introduce another attribute with `bin/sh` (and `usr/bin/env`) as a separate derivation, so we don't pull in a shell unconditionally.

Reply:
I included `var/empty` in `fakeNss`, and added a `dockerTools.binSh` derivation that provides `bin/sh`.